app/services/user_destroyer.rb

```diff
@@ -35,6 +35,10 @@ def destroy(user, opts = {})
       category_topic_ids = Category.where("topic_id IS NOT NULL").pluck(:topic_id)
 
       if opts[:delete_posts]
+        DiscoursePluginRegistry.user_destroyer_on_content_deletion_callbacks.each do |cb|
```

Comment:
Maybe:

```
DiscoursePluginRegistry.before_user_posts_deletion_callbacks
```

And then we go with:

```
register_before_user_posts_deletion_callback
```

I think it's a bit more explicit?

Reply:
I wanted to emphasize that the content deletion only happens while destroying the user, so I think your suggestion is actually less explicit.